refactor: correct format string warnings for size_t and uint64_t types

vjardin opened the pull request:
Fix compiler warnings about format specifier mismatches:

- Use %zu instead of %lu for size_t variables in:
  * btlv_enc.c - bytes_needed, len parameters
  * df_name.c - DF_NAME length, record numbers
  * fs.c - record_no, len parameters
  * fs_utils.c - record iteration index
  * softsim.c - rsp_len, request_len
  * storage.c - read/write offsets
  * uicc_remote_cmd.c - written_length, commands_len

- Use PRIu64/PRIx64 from <inttypes.h> for uint64_t variables in:
  * uicc_remote_cmd.c - CNTR counter values

- Change ss_file_size() signature from char* to const char*
  * fs.h - consistent with other path functions in the API

vjardin committed:
fix: correct format string warnings for size_t and uint64_t types

Fix compiler warnings about format specifier mismatches:

- Use %zu instead of %lu for size_t variables in:
  * btlv_enc.c - bytes_needed, len parameters
  * df_name.c - DF_NAME length, record numbers
  * fs.c - record_no, len parameters
  * fs_utils.c - record iteration index
  * softsim.c - rsp_len, request_len
  * storage.c - read/write offsets
  * uicc_remote_cmd.c - written_length, commands_len

- Use PRIu64/PRIx64 from <inttypes.h> for uint64_t variables in:
  * uicc_remote_cmd.c - CNTR counter values

- Change ss_file_size() signature from char* to const char*
  * fs.h - consistent with other path functions in the API

Signed-off-by: Vincent Jardin <vjardin@free.fr>

vjardin committed:
fix: correct format string warnings for size_t and uint64_t types

Fix compiler warnings about format specifier mismatches:

- Use %zu instead of %lu for size_t variables in:
  * btlv_enc.c - bytes_needed, len parameters
  * df_name.c - DF_NAME length, record numbers
  * fs.c - record_no, len parameters
  * fs_utils.c - record iteration index
  * softsim.c - rsp_len, request_len
  * storage.c - read/write offsets
  * uicc_remote_cmd.c - written_length, commands_len

- Use PRIu64/PRIx64 from <inttypes.h> for uint64_t variables in:
  * uicc_remote_cmd.c - CNTR counter values

- Change ss_file_size() signature from char* to const char*
  * fs.h - consistent with other path functions in the API

Signed-off-by: Vincent Jardin <vjardin@free.fr>

vjardin commented:
Please @benjaminbruun ?

benjaminbruun commented:
@vjardin thank you for the contributions! 🚀

sorry about the delay here, but we will get to them

We are going over some internal things and should very soon be ready to process these request. In the meantime, you are always welcome to keep these changes in your fork - but it would of course be beneficial for everyone once merged to mainline.

benjaminbruun commented:
Running some test on the nRF9151 DK.
Have you experienced issues other than the warnings when using the z (size_t)?
I would prefer to use PRI or maybe cast to a fixed-width integer to eliminate z (size_t).

vjardin commented:
> Running some test on the nRF9151 DK. Have you experienced issues other than the warnings when using the z (size_t)? I would prefer to use PRI or maybe cast to a fixed-width integer to eliminate z (size_t).

All the issues I have found, I did report them here. Do you have something in mind I should double check ?

vjardin commented:
please @benjaminbruun : why isn't it moving forward ?

vjardin commented:
Any news ?

vjardin commented:
Same, comment that the other pull request,

Following the DM for the CLA from last week, is it ok now ?

vjardin commented:
Hi @benjaminbruun : since CLA is set, please,  what's the issue with this pull request ?

benjaminbruun commented:
Sat that this PR was hanging - let's get it wrapped up and merged asap.

benjaminbruun reviewed (changes requested):
The `const char *` change in `fs.h` wasn't carried through to the implementation — fs.c:59 still declares `int ss_file_size(char *path)`, which causes a conflicting types error at build time. Could you update `fs.c` to match?
`int ss_file_size(const char *path)`

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/onomondo/onomondo-uicc?pullRequest=37) <br/>All committers have signed the CLA.

benjaminbruun commented:
Would love for this to be part of release v2.2.0. Will see if I can add the little change to this PR

benjaminbruun committed:
fix: update ss_file_size implementation for const correctness

The signature change in fs.h from this PR was not carried through to
the implementation in src/softsim/fs.c, causing clang to report
"conflicting types for 'ss_file_size'" at build time.

Signed-off-by: Benjamin Bruun <bb@onomondo.com>
